public/app/plugins/panel/xychart/ManualEditor.tsx

```diff
@@ -101,23 +112,53 @@ export const ManualEditor = ({
       </div>
 
       {selected >= 0 && value[selected] && (
-        <ScatterSeriesEditor
-          key={`series/${selected}`}
-          baseNameMode={FieldNamePickerBaseNameMode.ExcludeBaseNames}
-          item={{} as StandardEditorsRegistryItem}
-          context={context}
-          value={value[selected]}
-          onChange={(v) => {
-            onChange(
-              value.map((obj, i) => {
-                if (i === selected) {
-                  return v!;
+        <>
+          {frameNames.length > 1 && (
+            <Field label={'Data'}>
+              <Select
+                isClearable={false}
+                options={frameNames}
+                placeholder={'Change filter'}
+                value={
+                  frameNames.find((v) => {
+                    return v.value === value[selected].frame;
+                  }) ?? 0
                 }
-                return obj;
-              })
-            );
-          }}
-        />
+                onChange={(v) => {
+                  onChange(
+                    value.map((obj, i) => {
+                      if (i === selected) {
+                        if (v === null) {
+                          return { ...value[i], frame: undefined };
+                        }
+                        return { ...value[i], frame: v?.value!, x: undefined, y: undefined };
+                      }
+                      return obj;
+                    })
+                  );
+                }}
+              />
+            </Field>
+          )}
+          <ScatterSeriesEditor
+            key={`series/${selected}`}
+            baseNameMode={FieldNamePickerBaseNameMode.ExcludeBaseNames}
+            item={{} as StandardEditorsRegistryItem}
```

Comment:
I thought type assertions were naughtynaughty? 🤔

Reply:
typically this object is used to provide any editor specific settings as well as the `id` which is used to persist the information in configs and `name` which can be changed and doesn't impact configs

in this case we don't have any specific settings for this editor and we embedding a custom editor within another custom editor (vs using builder api) so I think this is acceptable 🤷🏻‍♂️

the alternative would be something like `item={{ id: `series/${selected}`, name: `series/${selected}` }}` but I don't feel strongly either way on this